client/app/lib/components/ConfirmDialog.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { PropTypes } from 'react';
+import Dialog from 'material-ui/Dialog';
+import RaisedButton from 'material-ui/RaisedButton';
+
+const propTypes = {
+  message: PropTypes.string,
+  onConfirm: PropTypes.func,
+};
+
+const buttonStyle = {
+  margin: 12,
+};
+
+class ConfirmDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { open: true };
+  }
+
+  renderConfirmDialog(message) {
+    const actions = [
+      <RaisedButton
+        primary
+        keyboardFocused
+        style={buttonStyle}
+        label="Cancel"
+        onTouchTap={() => this.setState({ open: false })}
+      />,
+      <RaisedButton
+        primary
+        style={buttonStyle}
+        label="Submit"
+        onTouchTap={() => this.props.onConfirm()}
+      />,
+    ];
+
+    return (
+      <div>
+        <Dialog
+          modal
+          actions={actions}
+          open={this.state.open}
+        >
+          {message}
+        </Dialog>
+      </div>
+    );
+  }
+
+  render() {
+    const { message } = this.props;
+    return this.state.open ? this.renderConfirmDialog(message) : null;
```

Comment:
I think you should render the `Dialog` anyway, then the `state` to determine whether it shows or not.